connectors/agent/service_manager.py

```diff
@@ -0,0 +1,53 @@
+from es_agent_client.util.logger import logger
+
+from connectors.logger import DocumentLogger
+from connectors.services.base import ServiceAlreadyRunningError, get_services
+from connectors.utils import CancellableSleeps
+
+
+class ConnectorServiceManager:
+    name = "connector-service-manager"
+
+    def __init__(self, configuration):
+        service_name = self.__class__.name
+        self.logger = DocumentLogger(
+            f"[{service_name}]", {"service_name": service_name}
+        )
+        self.agent_config = configuration
+        self._multi_service = None
+        self.running = False
+        self._sleeps = CancellableSleeps()
+
+    async def run(self):
+        if self.running:
+            msg = f"{self.__class__.__name__} is already running."
+            raise ServiceAlreadyRunningError(msg)
+
+        self.running = True
+        try:
+            while self.running:
+                try:
+                    logger.info("Starting connector services")
+                    self._multi_service = get_services(
+                        ["schedule", "sync_content", "sync_access_control", "cleanup"],
+                        self.agent_config.get(),
+                    )
+                    await self._multi_service.run()
+                except Exception as e:
+                    logger.exception(
+                        f"Error while running services in ConnectorServiceManager: {e}"
+                    )
+                    raise
+        finally:
```

Comment:
should we care about stopping running services here?

Reply:
It should already be stopped - it's not possible to reach here without stopping. I can add additional step of verifying that everything stopped, but not sure it's needed